worlds/inscryption/Rules.py

```diff
@@ -0,0 +1,189 @@
+            "Act 3 - Shop Holo Pelt": self.has_gems_and_battery,  # Costs money, so can need multiple battles.
+            "Act 3 - Middle Holo Pelt": self.has_inspectometer_battery,  # Can be reached without but possible soft lock
+            "Act 3 - Forest Holo Pelt": self.has_inspectometer_battery,
+            "Act 3 - Crypt Holo Pelt": self.has_inspectometer_battery,
+            "Act 3 - Tower Holo Pelt": self.has_gems_and_battery,
+            "Act 3 - Trader 1": self.has_pelts(1),
+            "Act 3 - Trader 2": self.has_pelts(2),
+            "Act 3 - Trader 3": self.has_pelts(3),
+            "Act 3 - Trader 4": self.has_pelts(4),
+            "Act 3 - Trader 5": self.has_pelts(5),
+            "Act 3 - Goobert's Painting": self.has_gems_and_battery,
+            "Act 3 - The Great Transcendence": self.has_transcendence_requirements,
+            "Act 3 - Boss Mycologists": self.has_mycologists_boss_requirements,
+            "Act 3 - Bone Lord Room": self.has_bone_lord_room_requirements,
+            "Act 3 - Luke's File Entry 1": self.has_battery_and_quill,
+            "Act 3 - Luke's File Entry 2": self.has_battery_and_quill,
+            "Act 3 - Luke's File Entry 3": self.has_battery_and_quill,
+            "Act 3 - Luke's File Entry 4": self.has_transcendence_requirements,
+            "Act 3 - Well": self.has_inspectometer_battery,
+            "Act 3 - Gems Drone": self.has_inspectometer_battery,
+            "Act 3 - Clock": self.has_gems_and_battery,  # Can be brute-forced, but the solution needs those items.
+        }
+        self.region_rules = {
+            "Act 2": self.has_act2_requirements,
+            "Act 3": self.has_act3_requirements,
+            "Epilogue": self.has_epilogue_requirements
+        }
+
+    def has_wardrobe_key(self, state: CollectionState) -> bool:
+        return state.has("Wardrobe Key", self.player)
+
+    def has_caged_wolf(self, state: CollectionState) -> bool:
+        return state.has("Caged Wolf Card", self.player)
+
+    def has_dagger(self, state: CollectionState) -> bool:
+        return state.has("Dagger", self.player)
+
+    def has_magnificus_eye(self, state: CollectionState) -> bool:
+        return state.has("Magnificus Eye", self.player)
+
+    def has_useful_act1_items(self, state: CollectionState) -> bool:
+        return state.has("Oil Painting's Clover Plant", self.player) and state.has("Squirrel Totem Head", self.player)
+
+    def has_all_epitaph_pieces(self, state: CollectionState) -> bool:
+        if self.world.options.epitaph_pieces_randomization.value == EpitaphPiecesRandomization.option_all_pieces:
+            return state.has("Epitaph Piece", self.player, 9)
+        elif self.world.options.epitaph_pieces_randomization.value == EpitaphPiecesRandomization.option_in_groups:
+            return state.has("Epitaph Pieces", self.player, 3)
+        else:
+            return state.has("Epitaph Pieces", self.player, 1)
```

Comment:
Two things. First, you could do this instead (also, I don't think you need `.value`):
```suggestion
        if self.world.options.epitaph_pieces_randomization == EpitaphPiecesRandomization.option_all_pieces:
            return state.has("Epitaph Piece", self.player, 9)
        if self.world.options.epitaph_pieces_randomization == EpitaphPiecesRandomization.option_in_groups:
            return state.has("Epitaph Pieces", self.player, 3)
        return state.has("Epitaph Pieces", self.player, 1)
```
Second, this is checking options inside of an access rule, which may get checked hundreds of time, so the if chain can be expensive, it would be best to refactor this so that you're not checking options inside the rule, either through lambdas or assigning a value to instances of your world class and using that.